[New] Display points using clustering feature reduction (#238)
Esri/arcgis-maps-sdk-swift-samples: Shared/Samples/Display points using clustering feature reduction/README.md

## Use case

Feature clustering can be used to dynamically aggregate groups of points that are within proximity of each other in order to represent each group with a single symbol. Such grouping allows you to see patterns in the data that are difficult to visualize when a layer contains hundreds or thousands of points that overlap and cover each other.

## How to use the sample

Pan and zoom the map to view how clustering is dynamically updated. Toggle clustering off to view the original point features that make up the clustered elements. When clustering is toggled on, you can tap on a clustered geoelement to view aggregated information and summary statistics for that cluster. When clustering is toggled off and you tap on the original feature you get access to information about individual power plant features.

## How it works

1. Create a map from a web map `PortalItem`.
2. Get the cluster enabled layer from the map's operational layers.
3. Get the `FeatureReduction` from the feature layer and set `isEnabled` to enable or disable clustering on the feature layer.
4. Use the `onSingleTapGesture` modifier to listen for tap events on the map view.

> 5. Identify tapped features on the map using `identify(on layer:screenPoint:tolerance:)` on the feature layer and pass in the map screen point location.

Reviewer: ```suggestion
5. Identify tapped features on the map using `identify(on:screenPoint:tolerance:returnPopupsOnly:maximumResults:)` on the feature layer and pass in the map screen point location.
```

6. Get the `Popup` from the resulting `IdentifyLayerResult` and use it to construct a `PopupView`.
7. Use a `FloatingPanel` to display the popup information from the `PopupView`.

## Relevant API

* AggregateGeoElement
* FeatureLayer
* FeatureReduction
* GeoElement
* IdentifyLayerResult

## About the data

> This sample uses a [web map](https://www.arcgis.com/home/item.html?id=8916d50c44c746c1aafae001552bad23) that displays the [Esri Global Power](https://www.arcgis.com/home/item.html?id=eb54b44c65b846cca12914b87b315169) Plants feature layer with feature reduction enabled. When enabled, the aggregate features symbology shows the color of the most common power plant type, and a size relative to the average plant capacity of the cluster.

Reviewer: ```suggestion
This sample uses a [web map](https://www.arcgis.com/home/item.html?id=8916d50c44c746c1aafae001552bad23) that displays the [Esri Global Power Plants](https://www.arcgis.com/home/item.html?id=eb54b44c65b846cca12914b87b315169) feature layer with feature reduction enabled. When enabled, the aggregate features symbology shows the color of the most common power plant type, and a size relative to the average plant capacity of the cluster.
```

> This sample uses a [web map](https://www.arcgis.com/home/item.html?id=8916d50c44c746c1aafae001552bad23) that displays the [Esri Global Power Plants](https://www.arcgis.com/home/item.html?id=eb54b44c65b846cca12914b87b315169) feature layer with feature reduction enabled. When enabled, the aggregate features symbology shows the color of the most common power plant type, and a size relative to the average plant capacity of the cluster.

Reviewer 2: ```suggestion
This sample uses a [web map](https://www.arcgis.com/home/item.html?id=8916d50c44c746c1aafae001552bad23) that displays the [Esri Global Power Plants](https://www.arcgis.com/home/item.html?id=eb54b44c65b846cca12914b87b315169) feature layer with feature reduction enabled. When enabled, the cluster's symbology shows the color of the most common power plant type, and a size relative to the average plant capacity of the cluster.
```
The perspective of this suggestion is that the result of aggregating features is a cluster and that cluster is symbolized. If that's not the right way to think of it then don't us this suggestion. If you don't use it then "features" should probably be possessive "feature's"
